caps/repo.go

```diff
@@ -0,0 +1,164 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2015 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package caps
+
+import (
+	"fmt"
+	"sort"
+	"sync"
+)
+
+// Repository stores all known snappy capabilities and types
+type Repository struct {
+	// Protects the internals from concurrent access.
+	// If contention gets high, switch to a RWMutex
```

Comment:
Can we please just drop the prescriptive suggestion? If contention gets high, the very first thing to do is to understand why contention is high. If we have more writes than reads, for example, a RWMutex does no good.